docs/dev/table/streaming/match_recognize.md

```diff
@@ -0,0 +1,654 @@
+
+{% highlight text %}
+ symbol   lastPrice
+======== ===========
+ XYZ      13
+{% endhighlight %}
+
+<span class="label label-danger" style="font-size:75%">Note:</span> It is not possible to use greedy quantifier for the last
+variable for a pattern, thus pattern like `(A B*)` is not allowed. This can be easily worked around by introducing artificial state
+e.g. `C` that will have a negated condition of `B`. So you could use a query like:
+
+{% highlight sql %}
+PATTERN (A B* C)
+DEFINE
+    A as condA()
+    B as condB()
+    C as NOT condB()
+{% endhighlight %}
+
+<span class="label label-danger" style="font-size:75%">Note:</span> Right now optional reluctant quantifier (`A??` or `A{0,1}?`) is not supported.
+
+Output mode
+-----------
+
+Currently supported output mode is `ONE ROW PER MATCH` that will always produce on output summary row per each found match.
+The schema of the output row will be following union of `{partitioning columns} + {measures columns}` in that particular order.
+
+Example:
+
+Query:
+{% highlight sql %}
+SELECT * 
+FROM Ticker
+    MATCH_RECOGNIZE(
+        PARTITION BY symbol
+        ORDER BY rowtime
+        MEASURES
+            FIRST(A.price) as startPrice
+            LAST(A.price) as topPrice
+            B.price as lastPrice
+        PATTERN (A+ B)
+        ONE ROW PER MATCH
+        DEFINE
+            A as A.price > LAST(A.price, 1) OR LAST(A.price, 1) IS NULL,
+            B as B.price < LAST(A.price)
+    )
+{% endhighlight %}
+
+for input:
+
+{% highlight text %}
+ symbol   tax   price.         rowtime
+======== ===== ======== =====================
+ XYZ      1     10       2018-09-17 10:00:02
+ XYZ      2     12       2018-09-17 10:00:03
+ XYZ      1     13       2018-09-17 10:00:04
+ XYZ      2     11       2018-09-17 10:00:05
+{% endhighlight %}
+
+will produce:
+
+{% highlight text %}
+ symbol   startPrice   topPrice   lastPrice
+======== ============ ========== ===========
+ XYZ      10           13         11
+{% endhighlight %}
+
+Event stream navigation
+------------------
+
+### Pattern variable reference
+Pattern variable reference e.g. A.price describes a set of rows mapped so far to A plus current row, if we try to match current row to A. If the expression requires a single row e.g. A.price > 10, A.price selects the last value of the row.
+
+If no pattern variable is specified e.g. SUM(price) it references the default pattern variable which references all variables in the pattern, so it constitutes row set of all rows mapped so far plus current row.
+
+Example:
+
+{% highlight sql %}
+...
+PATTERN (A B+)
+DEFINE
+  A as A.price > 10,
+  B as B.price > A.price AND SUM(price) < 100 AND SUM(B.price) < 80
+{% endhighlight %}
+  
+By `{A.price}` we describe set of rows that the values in DEFINE clause are evaluated on. The rows are referenced by its number
+
+<table class="table table-bordered">
```

Comment:
Use markdown syntax instead?

Reply:
I tried it, but I couldn't make it bordered and found it harder to read, moreover for the next table I control which rows can be wrapped. I think this is not supported in the markdown syntax as well, but I wanted to keep the same style for both tables.